src/main/java/com/artipie/jfr/JfrStorage.java

```diff
@@ -37,136 +45,230 @@ public JfrStorage(final Storage original) {
 
     @Override
     public CompletableFuture<Boolean> exists(final Key key) {
+        final CompletableFuture<Boolean> res;
         final StorageExistsEvent event = new StorageExistsEvent();
-        event.storage = this.identifier();
-        event.key = key.string();
-        event.begin();
-        return this.original.exists(key)
-            .thenApply(
-                res -> {
-                    event.commit();
-                    return res;
-                }
-            );
+        if (event.isEnabled()) {
+            event.begin();
+            res = this.original.exists(key)
+                .thenApply(
+                    rt -> this.eventProcess(rt, key, event, JfrStorage.EMPTY_RUNNABLE)
```

Comment:
Could we use more sensible name for 'rt' variable, for example 'exists'?

Reply:
ok, done